cpu/cc26x0/periph/timer.c

```diff
@@ -82,84 +133,120 @@ int timer_set(tim_t tim, int channel, unsigned int timeout)
+            dev(tim)->TBMATCHR = (timer_config[tim].cfg == GPT_CFG_32T) ?
+                                 value : (LOAD_VALUE - value);
+            dev(tim)->IMR |= GPT_IMR_TBMIM;
+            break;
+    }
+    return 1;
 }
 
 int timer_clear(tim_t tim, int channel)
 {
-    if (channel != 0) {
+    if ((tim >= TIMER_NUMOF) || (channel >= timer_config[tim].chn)) {
         return -1;
     }
 
-    dev(tim)->IMR &= ~(GPT_IMR_TAMIM);
+    dev(tim)->IMR &= (channel == 0) ? ~(GPT_IMR_TAMIM) : ~(GPT_IMR_TBMIM);
 
     return 0;
 }
 
 unsigned int timer_read(tim_t tim)
 {
-    return dev(tim)->TAV & 0xFFFF;
+    if (tim >= TIMER_NUMOF) {
+        return 0;
+    }
+    if (timer_config[tim].cfg == GPT_CFG_32T) {
+        return dev(tim)->TAV;
+    }
+    return LOAD_VALUE - (dev(tim)->TAV & 0xFFFF);
 }
 
 void timer_stop(tim_t tim)
 {
-    dev(tim)->CTL = 0;
+    if (tim < TIMER_NUMOF) {
+        dev(tim)->CTL = 0;
+    }
 }
 
 void timer_start(tim_t tim)
 {
-    dev(tim)->CTL = GPT_CTL_TAEN;
+    if (tim < TIMER_NUMOF) {
+        switch (timer_config[tim].chn) {
+            case 1:
+                dev(tim)->CTL = GPT_CTL_TAEN;
+                break;
+            case 2:
+                dev(tim)->CTL = GPT_CTL_TAEN | GPT_CTL_TBEN;
+                break;
+        }
+    }
 }
 
 /**
- * @brief           handle interrupt for a timer
+ * @brief   handle interrupt for a timer channel 0 (A)
  *
  * @param[in] tim   index of the timer
  */
-static inline void isr_handler(tim_t tim)
-{
-    uint32_t mis = dev(tim)->MIS;
+static void irq_handler_a(tim_t tim) {
+    uint32_t mis;
+    /* Latch the active interrupt flags */
+    mis = dev(tim)->MIS & TIMER_A_IRQ_MASK;
+    /* Clear the latched interrupt flags */
     dev(tim)->ICLR = mis;
 
     if (mis & GPT_IMR_TAMIM) {
+        /* Disable further match interrupts for this timer/channel */
         dev(tim)->IMR &= ~GPT_IMR_TAMIM;
+        /* Invoke the callback function */
         ctx[tim].cb(ctx[tim].arg, 0);
     }
 
     cortexm_isr_end();
 }
 
-#ifdef TIMER_0_ISR
-void TIMER_0_ISR(void)
-{
-    isr_handler(0);
-}
-#endif
+/**
+ * @brief   handle interrupt for a timer channel 1 (B)
+ *
+ * @param[in] tim   index of the timer
+ */
+static void irq_handler_b(tim_t tim) {
+    uint32_t mis;
+    /* Latch the active interrupt flags */
+    mis = dev(tim)->MIS & TIMER_B_IRQ_MASK;
+    /* Clear the latched interrupt flags */
+    dev(tim)->ICLR = mis;
 
-#ifdef TIMER_1_ISR
-void TIMER_1_ISR(void)
-{
-    isr_handler(1);
-}
-#endif
+    if (mis & GPT_IMR_TBMIM) {
+        /* Disable further match interrupts for this timer/channel */
+        dev(tim)->IMR &= ~GPT_IMR_TBMIM;
+        /* Invoke the callback function */
+        ctx[tim].cb(ctx[tim].arg, 1);
+    }
 
-#ifdef TIMER_2_ISR
-void TIMER_2_ISR(void)
-{
-    isr_handler(2);
+    cortexm_isr_end();
 }
-#endif
 
-#ifdef TIMER_3_ISR
-void TIMER_3_ISR(void)
-{
-    isr_handler(3);
-}
-#endif
+void isr_timer0_chan0(void) {irq_handler_a(0);}
```

Comment:
As said offline: I basically took `cpu/cc2538/periph/timer.c` as a template - still, there my be better ways to implement that.